swap/swap_test.go

```diff
@@ -449,6 +452,55 @@ func TestNewSwapFailure(t *testing.T) {
 	}
 }
 
+//TestStartChequebookWithParamAndPreviouslyOtherUsedChequebook tests that we cannot connect to another instance of a chequebook contract if on the same networkID we have already saved an instance of a chequebook
+func TestStartChequebookWithParamAndPreviouslyOtherUsedChequebook(t *testing.T) {
+	var expectedError = fmt.Errorf("Attempting to connect to provided chequebook, but different chequebook used before at networkID %s", testNetworkID)
+	swap, clean := newTestSwap(t, ownerKey)
+	defer clean()
+	// deploy
+	err := testDeploy(context.TODO(), swap)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//save
+	swap.saveChequebook(swap.GetParams().ContractAddress)
+	// we cannot give a flag, while on the same networkID we have already saved another instance of the chequebook
+	err = swap.StartChequebook(testChequeContract2)
+	if err.Error() != expectedError.Error() {
+		t.Fatalf("Error is not equal to expected error. Error: %s, Expected: %s", err, expectedError)
+	}
+}
+
+//TestStartChequebookNoChequebookParamAndPreviouslyOtherUsedChequebook tests that we cannot connect to a chequebook that is not a chequebook
+func TestStartChequebookNoChequebookParamAndPreviouslyOtherUsedChequebook(t *testing.T) {
```

Comment:
consider shortening these names if possible

Reply:
Sure thing. I will fix it tomorrow.